hrm-service/unit-tests/contact/contact-data-access.test.ts

```diff
@@ -0,0 +1,276 @@
+import * as pgPromise from 'pg-promise';
+import { mockConnection, mockTask, mockTransaction } from '../mock-pgpromise';
+import { search, create } from '../../src/contact/contact-data-access';
+import { endOfDay, startOfDay } from 'date-fns';
+import { ContactBuilder } from './contact-builder';
+import { NewContactRecord, insertContactSql } from '../../src/contact/sql/contact-insert-sql';
+
+jest.mock('../../src/contact/sql/contact-insert-sql', () => ({
+  insertContactSql: jest.fn().mockReturnValue('MOCKED INSERT STATEMENT'),
+}));
+
+let conn: pgPromise.ITask<unknown>;
+
+beforeEach(() => {
+  conn = mockConnection();
+});
+
+describe('create', () => {
+  const sampleNewContact: NewContactRecord = {
+    rawJson: {
+      childInformation: {
+        name: {
+          firstName: 'Lorna',
+          lastName: 'Ballantyne',
+        },
+      },
+      callType: 'carrier pigeon',
+      caseInformation: {
+        categories: {},
+      },
+    },
+    queueName: 'Q',
+    conversationDuration: 100,
+    twilioWorkerId: undefined,
+    timeOfContact: undefined,
+    createdBy: undefined,
+    helpline: undefined,
+    taskId: undefined,
+    channel: undefined,
+    number: undefined,
+    channelSid: undefined,
+    serviceSid: undefined,
+  };
+
+  test('No task ID specified in payload - runs SQL to insert new record and connect CSAM reports, using current date for created / updated and accountSid parameter', async () => {
```

Comment:
I wonder if this should be valid, since `taskId` is our way to guarantee idempotence, maybe it should be non nullable..

Reply:
Not sure? The logic I migrated seemed to support a case where no taskId was provided so I preserved that. If we don't forsee cases where contacts could be created without tasks, we could prohibit this use case